DUI3-DX/Converters/Rhino/Speckle.Converters.Rhino7/Geometry/BrepToSpeckleConverter.cs

```diff
@@ -0,0 +1,200 @@
+      curves2d = target.Curves2D.Select(curve2d => _curveConverter.RawConvert(curve2d)).ToList();
+    }
+
+    var speckleBrep = new SOG.Brep
+    {
+      Vertices = vertices,
+      Curve3D = curves3d,
+      Curve2D = curves2d,
+      Surfaces = surfaces,
+      displayValue = displayValue,
+      IsClosed = target.IsSolid,
+      Orientation = (SOG.BrepOrientation)target.SolidOrientation,
+      volume = target.IsSolid ? target.GetVolume() : 0,
+      area = target.GetArea(),
+      bbox = _boxConverter.RawConvert(new RG.Box(target.GetBoundingBox(false))),
+      units = _contextStack.Current.SpeckleUnits
+    };
+
+    // Brep non-geometry types
+    var faces = ConvertBrepFaces(target, speckleBrep);
+    var edges = ConvertBrepEdges(target, speckleBrep);
+    var loops = ConvertBrepLoops(target, speckleBrep);
+    var trims = ConvertBrepTrims(target, speckleBrep);
+
+    return speckleBrep;
+  }
+
+  private static List<SOG.BrepFace> ConvertBrepFaces(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Faces
+      .Select(
+        f =>
+          new SOG.BrepFace(
+            speckleParent,
+            f.SurfaceIndex,
+            f.Loops.Select(l => l.LoopIndex).ToList(),
+            f.OuterLoop.LoopIndex,
+            f.OrientationIsReversed
+          )
+      )
+      .ToList();
+
+  private List<SOG.BrepEdge> ConvertBrepEdges(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Edges
+      .Select(
+        edge =>
+          new SOG.BrepEdge(
+            speckleParent,
+            edge.EdgeCurveIndex,
+            edge.TrimIndices(),
+            edge.StartVertex?.VertexIndex ?? -1,
+            edge.EndVertex?.VertexIndex ?? -1,
+            edge.ProxyCurveIsReversed,
+            _intervalConverter.RawConvert(edge.Domain)
+          )
+      )
+      .ToList();
+
+  private List<SOG.BrepTrim> ConvertBrepTrims(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Trims
+      .Select(trim =>
+      {
+        var t = new SOG.BrepTrim(
+          speckleParent,
+          trim.Edge?.EdgeIndex ?? -1,
+          trim.Face.FaceIndex,
+          trim.Loop.LoopIndex,
+          trim.TrimCurveIndex,
+          (int)trim.IsoStatus,
+          (SOG.BrepTrimType)trim.TrimType,
+          trim.IsReversed(),
+          trim.StartVertex.VertexIndex,
+          trim.EndVertex.VertexIndex
+        )
+        {
+          Domain = _intervalConverter.RawConvert(trim.Domain)
+        };
+
+        return t;
+      })
+      .ToList();
+
+  private List<SOG.BrepLoop> ConvertBrepLoops(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Loops
+      .Select(
+        loop =>
+          new SOG.BrepLoop(
+            speckleParent,
+            loop.Face.FaceIndex,
+            loop.Trims.Select(t => t.TrimIndex).ToList(),
+            (SOG.BrepLoopType)loop.LoopType
+          )
+      )
+      .ToList();
+
+  private RG.Mesh? GetBrepDisplayMesh(RG.Brep brep)
```

Comment:
Yeah some of this seems to be Convert To logic and maybe could use some TLC when coming over, i.e the commented out switcg and settings.

We can probably accept this stuff as-us if we maybe articulate some steps to tidy with a CNX refactor story (not task as it need review and other bits a task may not need some much DOD applicability).

Reply:
Oh, this is missing a POC comment. It should be included in the context somehow, but it's not critical for conversion success so I left it out